owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/status/GetRemoteCapabilitiesOperation.kt

```diff
@@ -0,0 +1,323 @@
+ *   copies of the Software, and to permit persons to whom the Software is
+ *   furnished to do so, subject to the following conditions:
+ *
+ *   The above copyright notice and this permission notice shall be included in
+ *   all copies or substantial portions of the Software.
+ *
+ *   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *   EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ *   MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *   NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ *   BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ *   ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ *   CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *   THE SOFTWARE.
+ *
+ */
+
+package com.owncloud.android.lib.resources.status
+
+import com.owncloud.android.lib.common.OwnCloudClient
+import com.owncloud.android.lib.common.http.HttpConstants
+import com.owncloud.android.lib.common.http.methods.nonwebdav.GetMethod
+import com.owncloud.android.lib.common.operations.RemoteOperation
+import com.owncloud.android.lib.common.operations.RemoteOperationResult
+import com.owncloud.android.lib.common.operations.RemoteOperationResult.ResultCode.OK
+import com.owncloud.android.lib.common.utils.Log_OC
+import org.json.JSONObject
+import java.net.URL
+
+/**
+ * Get the Capabilities from the server
+ * Save in Result.getData in a RemoteCapability object
+ *
+ * @author masensio
+ * @author David González Verdugo
+ */
+/**
+ * Constructor
+ */
+class GetRemoteCapabilitiesOperation : RemoteOperation<RemoteCapability>() {
+
+    override fun run(client: OwnCloudClient): RemoteOperationResult<RemoteCapability> {
+        var result: RemoteOperationResult<RemoteCapability>
+
+        try {
+            val requestUri = client.baseUri
+            val uriBuilder = requestUri.buildUpon()
+            uriBuilder.appendEncodedPath(OCS_ROUTE)    // avoid starting "/" in this method
+            uriBuilder.appendQueryParameter(PARAM_FORMAT, VALUE_FORMAT)
+
+            val getMethod = GetMethod(URL(uriBuilder.build().toString()))
+
+            getMethod.addRequestHeader(RemoteOperation.OCS_API_HEADER, RemoteOperation.OCS_API_HEADER_VALUE)
+
+            val status = client.executeHttpMethod(getMethod)
+
+            val response = getMethod.responseBodyAsString
+            if (isSuccess(status)) {
+                Log_OC.d(TAG, "Successful response: " + response!!)
+
+                // Parse the response
+                val respJSON = JSONObject(response)
+                val respOCS = respJSON.getJSONObject(NODE_OCS)
+                val respMeta = respOCS.getJSONObject(NODE_META)
+                val respData = respOCS.getJSONObject(NODE_DATA)
+
+                // Read meta
+                val statusProp = respMeta.getString(PROPERTY_STATUS).equals("ok", ignoreCase = true)
```

Comment:
that's "ok" could be moved to a constant and used in several places, couldn't it?

Reply:
Done